scalafix/rules/src/main/scala/com/dwolla/security/crypto/V04to05.scala

```diff
@@ -1,34 +1,105 @@
 package com.dwolla.security.crypto
 
 import scalafix.v1.*
+
+import scala.annotation.tailrec
 import scala.meta.*
 
 class V04to05 extends SemanticRule("com.dwolla.security.crypto.V04to05") {
 
-  override def fix(implicit doc: SemanticDocument): Patch = {
+  override def fix(implicit doc: SemanticDocument): Patch =
     doc.tree.collect {
       case t@Term.ApplyType.After_4_6_0(Term.Name("CryptoAlg"), Type.ArgClause(List(Type.Name(name)))) =>
         Patch.replaceTree(t, s"CryptoAlg.resource[$name]").atomic
       case t@Term.Apply.After_4_6_0(Term.Select(Term.Name(name), Term.Name("armor")), Term.ArgClause(List(), None)) =>
         Patch.replaceTree(t, s"$name.armor").atomic
-      case t@Term.Apply.After_4_6_0(Term.Name("tagChunkSize"), _) =>
-        Patch.renameSymbol(t.symbol, "ChunkSize")
-      case t@Term.Apply.After_4_6_0(
-        Term.Name("PosInt"),
-        Term.ArgClause(List(Lit.Int(posIntArg)), None)
-      ) => Patch.replaceTree(t, s"PosInt.unsafeFrom($posIntArg)").atomic
-      case t@
-        Term.Apply.After_4_6_0(
-          Term.Select(Term.Name(cryptoRName), Term.Name("encrypt")),
-          Term.ArgClause(
-            List(
-              Term.Name(keyName),
-              Term.Assign( Term.Name("fileName"), fileNameTerm )
-            ),
-            None
-          )
-        ) =>
-        Patch.replaceTree(t, s"$cryptoRName.encrypt(EncryptionConfig().withFileName($fileNameTerm), $keyName)")
+      case Term.Apply.After_4_6_0(Term.Select(Term.Name(_), fun@Term.Name("encrypt")), t@Term.ArgClause((keyName@Term.Name(_)) :: additionalArguments, None)) =>
+        migrateEncrypt(t, fun, additionalArguments, Some(keyName), offset = 1)
+      case Term.Apply.After_4_6_0(Term.Select(Term.Name(_), fun@Term.Name("encrypt")), t@Term.ArgClause(arguments, None)) =>
+        migrateEncrypt(t, fun, arguments, None, offset = 0)
+      case t@Term.Apply.After_4_6_0(Term.Name("tagChunkSize"), _) if !isEncryptAParent(t) =>
+        Patch.replaceToken(t.tokens.head, "ChunkSize")
     }.asPatch
+
+  @tailrec
+  private def isEncryptAParent(t: Tree): Boolean =
+    t.parent match {
+      case Some(Term.Apply.After_4_6_0(Term.Select(_, Term.Name("encrypt")), _)) => true
+      case Some(other) => isEncryptAParent(other)
+      case _ => false
+    }
+
+  private def migrateEncrypt(t: Term.ArgClause,
```

Comment:
I can't think of other cases that would need to be covered that aren't covered by this, so +1 from me.